### PR TITLE
3273 preview of 3 d items in dropzone

### LINKED ISSUE #3273: Preview of 3D items in dropzone
_When user mint a 3D nft, show the nft preview in the dropzone_

- `/rmrk/create`
- `/bsx/create`

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #3273 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
image

### CODE REVIEW

File: components/rmrk/Create/3DViewer/ThreeDViewer.vue [hunk 1]
```diff
@@ -0,0 +1,245 @@
+<template>
+  <div id="canvasContainer" class="canvasContainer"></div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Watch } from 'nuxt-property-decorator'
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/rmrk/Create/3DViewer/ThreeDViewer.vue:6]: Sure

File: components/rmrk/Create/3DViewer/ThreeDViewer.vue [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  DirectionalLight,
+  Light,
+  LinearToneMapping,
+  Object3D,
+  PerspectiveCamera,
+  Scene,
+  Vector3,
+  WebGLRenderer,
+  sRGBEncoding,
+} from 'three'
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/rmrk/Create/3DViewer/ThreeDViewer.vue:20]: Cool, I'll get on it. Thank you

[the remark this answers is not in the document]
Comment (daiagi) [reply components/rmrk/Create/3DViewer/ThreeDViewer.vue]: @preschian 
I've deleted my ThreeDViwer component and the three dependency
using the existing 'ModelMedia' component to display the model
the existing 'ar-button' in ModelMedia swallowed click event, preventing the file upload menu from popping up
to solve that I added a props 'withARButton' to ModelMedia, the prop default value is true, to avoid breaking previous behavior

waiting for you review
Thanks

File: components/rmrk/Create/DropUpload.vue [hunk 3]
```diff
@@ -39,10 +44,12 @@
 <script lang="ts">
 import { Component, Emit, Prop, Ref, Vue, Watch } from 'nuxt-property-decorator'
 import Tooltip from '@/components/shared/Tooltip.vue'
+import ModelMedia from '~~/components/media/type/ModelMedia.vue'
```
[the remark this answers is not in the document]
Comment (roiLeo) [reply components/rmrk/Create/DropUpload.vue]: It doesn't work for me either

Screenshot 2022-11-09 at 13-27-39 KodaDot Low fees and low carbon minting KodaDot - Kusama NFT Market Explorer

[the remark this answers is not in the document]
Comment (daiagi) [reply components/rmrk/Create/DropUpload.vue]: no.
it doesn't work yet
problem is chrome doesn't seem to support model/* types
so in places where we use DropUpload without accept prop it works (rmrk/create nft)

but where we want to restrict allowed file type such as in BaseCollectionForm it doesn't, because chrome doesn't work with model/* types
(rmrk/create collection, bsx/ create collection)

it works well in Firefox

I'm open to suggestions on how to overcome this issue please

[the remark this answers is not in the document]
Comment (daiagi) [reply components/rmrk/Create/DropUpload.vue]: @roiLeo ok then, if that's the expected behavior then all good

**Review by roiLeo — APPROVED** [review]
lgtm

**Comment by daiagi** [comment]
@roiLeo @preschian 
Hi guys.
now what? is the merge waiting on me?

**Comment by roiLeo** [comment]
pay 50

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 27.15 USD/KSM ~ 1.842 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0x3571bd4e3d15ff773702b05b6ed0d88abf236f7d8a59a5f8845e6bf6ab65dc2c](https://kusama.subscan.io/extrinsic/0x3571bd4e3d15ff773702b05b6ed0d88abf236f7d8a59a5f8845e6bf6ab65dc2c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by roiLeo** [comment]
Merged! feel free to grab another issue
